src/editor-page.jsx

```diff
@@ -45,8 +47,41 @@ var EditorPage = React.createClass({
             wasAnswered: false
         };
     },
+    
+    getItemEditorSearchCount: function() {
+        var count = 0;
+        
+        count += Util.countOccurences(this.props.question.content, this.state.searchString);
+        count += Util.countOccurences(this.props.answerArea.options.content, this.state.searchString);
+
+        return count;
+    },
+    
+    getCombinedHintsSearchCount: function() {
+        var count = 0;
+        
+        this.props.hints.forEach(hint => {
+            count += Util.countOccurences(hint.content, this.state.searchString);
+        });
+
+        return count;
+    },
 
     render: function() {
+        
+        var itemEditorSearchIndex = -1;
+        var combinedHintsSearchIndex = -1;
+
+        if (this.state.searchIndex !== -1) {
+            var itemEditorCount = this.getItemEditorSearchCount();
+            var combinedHintsCount = this.getCombinedHintsSearchCount();
+
+            if (this.state.searchIndex < itemEditorCount) {
+                itemEditorSearchIndex = this.state.searchIndex;
+            } else if (this.state.searchIndex < itemEditorCount + combinedHintsCount) {
+                combinedHintsSearchIndex = this.state.searchIndex - itemEditorCount;
+            }
+        }
```

Comment:
This figures out what the searchIndex would be for each component if we were only considering that single component.  It could definitely use some cleanup.